packages/core/src/Provider/RewardsProvider/types.ts

```diff
@@ -5,6 +5,11 @@ import { Range } from '@cardano-sdk/util';
 export interface EpochRewards {
   epoch: Cardano.EpochNo;
   rewards: Cardano.Lovelace;
+  /**
+   * The pool the stake address was delegated to when the reward is earned.
+   * Will be null for payments from the treasury or the reserves.
+   */
+  poolId: Cardano.PoolId | null;
```

Comment:
This makes it forward-compatible (techically), as old providers would return objects with this `undefined`
```suggestion
  poolId?: Cardano.PoolId;
```